components/homme/src/share/gllfvremap_mod.F90

```diff
@@ -0,0 +1,2253 @@
+    gfr%npi = max(2, nphys)
+    nphys2 = nphys*nphys
+
+    call gfr_init_w_gg(np, gfr%w_gg)
+    call gfr_init_w_gg(gfr%npi, gfr%w_sgsg)
+    call gfr_init_w_ff(nphys, gfr%w_ff)
+    call gfr_init_M_gf(np, nphys, gfr%M_gf, .true.)
+    gfr%g2f_remapd(:,:,:nphys2) = reshape(gfr%M_gf(:,:,:nphys,:nphys), (/np,np,nphys2/))
+    call gfr_init_M_gf(gfr%npi, nphys, gfr%M_sgf, .false.)
+    call gfr_init_R(gfr%npi, nphys, gfr%w_sgsg, gfr%M_sgf, R, tau)
+    call gfr_init_interp_matrix(gfr%npi, gfr%interp)
+    call gfr_init_f2g_remapd(gfr, R, tau)
+
+    allocate(gfr%fv_metdet(nphys2,nelemd), &
+         gfr%D_f(nphys,nphys,2,2,nelemd), gfr%Dinv_f(nphys,nphys,2,2,nelemd), &
+         gfr%qmin(nlev,max(1,qsize),nelemd), gfr%qmax(nlev,max(1,qsize),nelemd), &
+         gfr%phis(nphys2,nelemd), gfr%spherep_f(nphys,nphys,nelemd))
+    call gfr_init_fv_metdet(elem, gfr)
+    call gfr_init_geometry(elem, gfr)
+
+    if (nphys == 1 .and. gfr%boost_pg1) call gfr_pg1_init(gfr)
+  end subroutine gfr_init
+
+  subroutine gfr_finish()
+    ! Deallocate the internal gfr structure.
+
+    if (.not. allocated(gfr%fv_metdet)) return
+    deallocate(gfr%fv_metdet, gfr%D_f, gfr%Dinv_f, gfr%qmin, gfr%qmax, gfr%phis, &
+         gfr%spherep_f)
+  end subroutine gfr_finish
+
+  subroutine gfr_dyn_to_fv_phys_hybrid(hybrid, nt, hvcoord, elem, nets, nete, &
+       ps, phis, T, uv, omega_p, q)
+    ! Remap ps, phis, T, uv, omega_p, q from GLL to FV grids.
+
+    use dimensions_mod, only: nlev
+    use hybvcoord_mod, only: hvcoord_t
+    use element_ops, only: get_temperature
+    use physical_constants, only: p0, kappa
+
+    type (hybrid_t), intent(in) :: hybrid
+    integer, intent(in) :: nt
+    type (hvcoord_t), intent(in) :: hvcoord
+    type (element_t), intent(in) :: elem(:)
+    integer, intent(in) :: nets, nete
+    real(kind=real_kind), intent(inout) :: ps(:,:), phis(:,:), T(:,:,:), &
+         uv(:,:,:,:), omega_p(:,:,:), q(:,:,:,:)
+
+    real(kind=real_kind), dimension(np,np,nlev) :: dp, dp_fv, wr1, wr2, p, p_fv
+    real(kind=real_kind) :: qmin, qmax, ones(np,np)
+    integer :: ie, nf, nf2, ncol, qi, qsize
+
+    ones = one
+    nf = gfr%nphys
+    nf2 = nf*nf
+    ncol = nf*nf
+
+    qsize = size(q,3)
+    
+    do ie = nets,nete
+       call gfr_g2f_scalar(ie, elem(ie)%metdet, elem(ie)%state%ps_v(:,:,nt:nt), &
+            wr1(:,:,:1))
+       ps(:ncol,ie) = reshape(wr1(:nf,:nf,1), (/ncol/))
+
+       if (gfr%have_fv_topo_file_phis) then
+          phis(:ncol,ie) = gfr%phis(:,ie)
+       else
+          call gfr_dyn_to_fv_phys_topo_elem(elem, ie, phis(:,ie))
+       end if
+
+       call calc_dp(hvcoord, elem(ie)%state%ps_v(:,:,nt), dp)
+       call gfr_g2f_scalar(ie, elem(ie)%metdet, dp, dp_fv)
+
+       call get_temperature(elem(ie), wr2, hvcoord, nt)
+       call calc_p(hvcoord, elem(ie)%state%ps_v(:,:,nt), p)
+       call gfr_g2f_scalar(ie, elem(ie)%metdet, p, p_fv)
+       wr2 = wr2*(p/p0)**kappa
```

Comment:
Why is here Exner pressure? Is this code only for theta-l?

Reply:
No, it works for all dycores. The idea is to remap theta so that remap is adiabatic. theta_v would probably be even better, but I don't want to bring in qv.